src/kirby/components/modal/modal.component.ts

```diff
@@ -0,0 +1,146 @@
+import {  Component,
+          OnInit,
+          OnDestroy,
+          Input,
+          Output,
+          EventEmitter,
+          ViewChild,
+          ElementRef,
+          AfterViewChecked,
+        } from '@angular/core';
+
+@Component({
+  selector: 'kirby-modal',
+  templateUrl: './modal.component.html',
+  styleUrls: ['./modal.component.scss']
+})
+export class ModalComponent implements OnInit, OnDestroy, AfterViewChecked {
+  // State of the modal - called from another component, e.g. a button
+  private _isOpen = false;
+
+  @Input() size?: 'small' | 'medium' | 'large';
+  // Theme option - e.g. dark, light or whatever theme there is
+  @Input() theme?: string;
+  modalCssClasses: {};
+  // Output closing state
+  @Output() closingModal = new EventEmitter();
```

Comment:
Should be named same way as click, tap, open, close:
```suggestion
@Output() close = new EventEmitter();
```